task17/index.js

```diff
@@ -0,0 +1,115 @@
+const URL = "https://mocki.io/v1/af7d1b29-da7d-4eba-977b-cdb5caeb2fff";
+
+const slideShowContainer = document.getElementById("js-slideShowContainer");
+const slideShowWrapper = document.getElementById("js-slideShowWrapper");
+const slideImgList = document.getElementById("js-slideImgList");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const addLoading = () => {
+  const loadingPlace = createElementWithClassName("div", "loader");
+  loadingPlace.id = "js-loadingPlace";
+  slideShowWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("js-loadingPlace").remove();
+
+const renderErrorMessage = (errorMessage) => {
+  slideShowWrapper.textContent = errorMessage;
+};
+
+const getFetchData = async () => {
+  const response = await fetch(URL);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage("問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImgData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(URL));
+    }, 3000);
+  });
+};
+
+const init = async () => {
+  addLoading();
+  try {
+    const value = await fetchImgData();
+    renderSlideImage(value);
+    renderSlideNumber(value);
+    addEventListenerForNextBtn(value);
+    addEventListenerForPrevBtn(value);
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
+  } finally {
+    removeLoading();
+  }
+};
+init();
+
+const renderSlideImage = (value) => {
+  const slideImgFragment = document.createDocumentFragment();
+  for (let i = 0; i < value.length; i++) {
+    const slideImgItem = createElementWithClassName("li", "slide-img-item");
+    const slideImage = createElementWithClassName("img", "slide-image");
+    slideImgItem.id = `js-slideImgItem${i}`;
+    slideImage.id = `slide-image${i}`;
+    slideImage.src = value[i].img;
+    slideImage.alt = value[i].alt;
+    value[i].display && slideImgItem.classList.add("is-active");
+    slideImgFragment.appendChild(slideImgItem).appendChild(slideImage);
+  }
+  slideShowWrapper.appendChild(slideImgList).appendChild(slideImgFragment);
```

Comment:
IMO
The slideImgList is marked up as a child element of the slideShowWrappe in HTML.
```suggestion
  slideImgList.appendChild(slideImgFragment);
```

Reply:
I didn't even notice until you mentioned it! Thank you. That was very refreshing!
9d03f7a6252ace723d8ae4f94806e9360c9513fa